### ISSUE TITLE
Make CCID-protected items discoverable but not downloadable for non-CCID users

### ISSUE THREAD

**Issue by pbinkley** [ualbertalib/HydraNorth#717]
This subsumes #710 #582 

CCID-protected items should be indexed and discoverable and displayable for all users, but the download should only work for CCID-authenticated users. Others should be redirected to the login page.

**Comment by sfbetz** [comment]
After conversation with Sharon F - some concern that if it takes a while to get this up and running, there may be cases where people are faced with this page when linking in from catalogue, libguides, etc.


![screen shot 2015-10-13 at 2 29 25 pm](https://cloud.githubusercontent.com/assets/12801010/10467472/235a7b56-71b7-11e5-995c-24d56295edb9.png)

Can we update the banner to the text? Or does is show for private items as well and would cause confusion? 

Suggested wording:
"This item is restricted to users with a University of Alberta CCID. Please log in with your CCID to view this item"

**Comment by weiweishi** [comment]
@sfarnel @anayram @pbinkley could you please generate a list of all the ccid-protected items, and this issue can be fixed by update the read_groups value on all the migrated items.

**Comment by anayram** [comment]
You can use the list available at:

Digital_Init:\Hydra_DAMS\ERA\ERA_migration\FOXML_transformed\complete_remigration\pid_lists\access_testing\ua_access

This list is generated against the migration package and follows the
conditions:
is CCID protected, is active, is not embargoed.

Let me know if you neeed something different.

On Wed, Oct 14, 2015 at 2:09 PM, Weiwei Shi <notifications@github.com>
wrote:

> @sfarnel <https://github.com/sfarnel> @anayram
> <https://github.com/anayram> @pbinkley <https://github.com/pbinkley>
> could you please generate a list of all the ccid-protected items, and this
> issue can be fixed by update the read_groups value on all the migrated
> items.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ualbertalib/HydraNorth/issues/717#issuecomment-148181148>
> .
>



-- 
Mariana Paredes-Olea
paredeso@ualberta.ca
780-492-7075

**Comment by mbarnett** [comment]
@weiweishi handing this off to you as discussed. I've fixed this issue for all new files on https://github.com/ualbertalib/HydraNorth/tree/717-fix-ccid-permissions

All you'll have to do in the rake task is:

```ruby
file.visibility = Hydranorth::AccessControls::InstitutionalVisibility::UNIVERSITY_OF_ALBERTA
file.save
```

for every existing CCID protected file. This will reset the read_groups from ["registered", "university_of_alberta"] to ["public", "university_of_alberta"]

**Comment by pbinkley** [comment]
Closed by #743
